### DISCUSSION TITLE
x509: certificate has expired or is not yet valid

### DISCUSSION THREAD

**Discussion by darrylmorley** [caprover/caprover#1749] [Q&A] [answered]
Hi,

I just tried to deploy an update to my site tonight but the update failed with the message `x509: certificate has expired or is not yet valid`. I'm using Caprover with a self-hosted docker registry.

Does anyone know what I need to do to fix it please?

**Comment by githubsaturn** [comment, marked as answer]
Try restarting your self hosted registry container. 
```
docker service update captain-registry --force
```

does it help?

**Reply by darrylmorley** [reply]
This solved the issue. Thank you. 👍
